src/Functions/FunctionMathBinaryFloat64.h

```diff
@@ -53,35 +53,89 @@ class FunctionMathBinaryFloat64 : public IFunction
+
+        const auto left_rows_remaining = left_src_size % Impl::rows_per_iteration;
+        const auto right_rows_remaining = right_src_size % Impl::rows_per_iteration;
+
+        const auto left_rows_size = left_src_size - left_rows_remaining;
+        const auto right_rows_size = right_src_size - right_rows_remaining;
+
+        const auto rows_size = std::max(left_rows_size, right_rows_size);
+
+        for (size_t i = 0; i < rows_size; i += Impl::rows_per_iteration)
+            Impl::execute(&left_src_data[i % left_rows_size], &right_src_data[i % right_rows_size], &dst_data[i]);
+
+        if (left_rows_remaining != 0 || right_rows_remaining != 0)
         {
-            auto dst = ColumnVector<Float64>::create();
+            LeftType left_src_remaining[Impl::rows_per_iteration];
+            memcpy(left_src_remaining, &left_src_data[left_rows_size % left_src_size], left_rows_remaining * sizeof(LeftType));
+            memset(left_src_remaining + left_rows_remaining, 0, (Impl::rows_per_iteration - left_rows_remaining) * sizeof(LeftType));
 
-            LeftType left_src_data[Impl::rows_per_iteration];
-            std::fill(std::begin(left_src_data), std::end(left_src_data), left_arg->template getValue<LeftType>());
-            const auto & right_src_data = right_arg_typed->getData();
-            const auto src_size = right_src_data.size();
-            auto & dst_data = dst->getData();
-            dst_data.resize(src_size);
+            RightType right_src_remaining[Impl::rows_per_iteration];
+            memcpy(right_src_remaining, &right_src_data[right_rows_size % right_src_size], right_rows_remaining * sizeof(RightType));
+            memset(right_src_remaining + right_rows_remaining, 0, (Impl::rows_per_iteration - right_rows_remaining) * sizeof(RightType));
 
-            const auto rows_remaining = src_size % Impl::rows_per_iteration;
-            const auto rows_size = src_size - rows_remaining;
+            Float64 dst_remaining[Impl::rows_per_iteration];
 
-            for (size_t i = 0; i < rows_size; i += Impl::rows_per_iteration)
-                Impl::execute(left_src_data, &right_src_data[i], &dst_data[i]);
+            Impl::execute(left_src_remaining, right_src_remaining, dst_remaining);
 
-            if (rows_remaining != 0)
-            {
-                RightType right_src_remaining[Impl::rows_per_iteration];
-                memcpy(right_src_remaining, &right_src_data[rows_size], rows_remaining * sizeof(RightType));
-                memset(right_src_remaining + rows_remaining, 0, (Impl::rows_per_iteration - rows_remaining) * sizeof(RightType));
-                Float64 dst_remaining[Impl::rows_per_iteration];
+            const auto rows_remaining = std::max(left_rows_remaining, right_rows_remaining);
 
-                Impl::execute(left_src_data, right_src_remaining, dst_remaining);
+            memcpy(&dst_data[rows_size], dst_remaining, rows_remaining * sizeof(Float64));
```

Comment:
Eventually, I left this code as if was before, since Impl::execute operates by Impl::rows_per_operation size blocks, which may go beyond dst_data boundary.